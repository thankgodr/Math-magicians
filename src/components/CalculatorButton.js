import React from 'react';
import PropTypes from 'prop-types';

export default class CalculatorButton extends React.Component {
  constructor(props) {
    super(props);
    this.clickHandler = this.clickHandler.bind(this);
  }

  clickHandler() {
    this.props.add(this.props.name);
  }

  render() {
    return (
            <div className={this.props.className} onClick={this.clickHandler} id={this.props.name}>
               <p className='text-center'> {this.props.name}</p>
            </div>
    );
  }
}

CalculatorButton.defaultProps = {
  name: '0',
  className: 'col-md-3 btn_number',
};

CalculatorButton.propTypes = {
  name: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
  add: PropTypes.func,
};