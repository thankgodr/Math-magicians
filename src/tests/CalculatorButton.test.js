import renderer from 'react-test-renderer';
import { render, queryByAttribute } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import CalculatorButton from '../components/CalculatorButton';

describe('Testing interaction and snaption of calculator button', () => {
  test('Calculator button snapshot', () => {
    const elem = renderer.create(<CalculatorButton />).toJSON();
    expect(elem).toMatchSnapshot();
  });

  test('Button is clickable', () => {
    let count = 0;
    const res = () => { count += 1; };
    const dom = render(<CalculatorButton name="2" add={res} clclassName = "tess" />);
    const getById = queryByAttribute.bind(null, 'id');
    const first = getById(dom.container, '2');
    userEvent.click(first);
    userEvent.click(first);
    expect(count).toBe(2);
  });
});