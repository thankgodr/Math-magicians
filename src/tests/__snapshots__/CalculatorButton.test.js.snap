// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing interaction and snaption of calculator button Calculator button snapshot 1`] = `
<div
  className="col-md-3 btn_number"
  id="0"
  onClick={[Function]}
>
  <p
    className="text-center"
  >
     
    0
  </p>
</div>
`;
